### PR TITLE
[hpal] add Tyr's Deliverance

### PR DESCRIPTION
**Pull request by mfield**
Open to suggestions on what/how to present this info. e.g. was unsure between aggregate or split out direct/buffed healing in the main statistic.

image
image
https://www.warcraftlogs.com/reports/L743dxBZTW1nXY6w#fight=11&type=healing&source=20

### CODE REVIEW

File: src/analysis/retail/paladin/holy/modules/talents/TyrsDeliverance.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        ])
+        .by(SELECTED_PLAYER),
+      this.onBuffedHeal,
+    );
+  }
+
+  onApply(event: ApplyBuffEvent) {
+    this.casts += 1;
+    this.lastApply = event.timestamp;
+    console.log('GLIMDEBUG apply buff', this.owner.formatTimestamp(event.timestamp));
```
Comment (abelito75) [inline src/analysis/retail/paladin/holy/modules/talents/TyrsDeliverance.tsx]: No console.logs in prod

File: src/analysis/retail/paladin/holy/modules/talents/TyrsDeliverance.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  }
+
+  onRemove(event: RemoveBuffEvent | FightEndEvent) {
+    if (!this.selectedCombatant.hasBuff(TALENTS.TYRS_DELIVERANCE_TALENT.id)) {
+      return;
+    }
+    console.log(
```
Comment (abelito75) [inline src/analysis/retail/paladin/holy/modules/talents/TyrsDeliverance.tsx]: No console.logs in prod

File: src/analysis/retail/paladin/holy/modules/talents/TyrsDeliverance.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+      Events.removebuff.spell([TALENTS.TYRS_DELIVERANCE_TALENT]).by(SELECTED_PLAYER),
+      this.onRemove,
+    );
+    this.addEventListener(Events.fightend, this.onRemove);
+
+    this.addEventListener(
+      Events.heal.spell([SPELLS.TYRS_DELIVERANCE_HEALING_INCREASE]).by(SELECTED_PLAYER),
+      this.onDirectHeal,
+    );
+    this.addEventListener(
```
Comment (abelito75) [inline src/analysis/retail/paladin/holy/modules/talents/TyrsDeliverance.tsx]: Would be nice to see this per spell

**Review by abelito75 — CHANGES_REQUESTED** [review]
This seems to not divide out Boundless Salvation from the base Tyrs cast... I thought we wanted to be able to see both

File: src/analysis/retail/paladin/holy/modules/talents/TyrsDeliverance.tsx (hunk 1 above)
Comment (mfield) [reply src/analysis/retail/paladin/holy/modules/talents/TyrsDeliverance.tsx]: whoops :(

File: src/analysis/retail/paladin/holy/modules/talents/TyrsDeliverance.tsx (hunk 3 above)
Comment (mfield) [reply src/analysis/retail/paladin/holy/modules/talents/TyrsDeliverance.tsx]: image

**Comment by mfield** [comment]
> This seems to not divide out Boundless Salvation from the base Tyrs cast... I thought we wanted to be able to see both

I'm not sure what you had in mind here.  For me, seeing how much I extended my Tyr's buff on average is enough for now. Can we call this good enough for a first pass and extend it later if we want to try and dive into something like "why'd you heal Alice when Bob had the buff and could have given you more extension?"

**Comment by abelito75** [comment]
My brain was just 
Base 20 seconds of Tyrs goes to the Tyrs box so from 0 - 20
Extended 40 seconds goes to the Boundless box so from 20 - 60

No like `why didn't you extend it longer` or anything

Since you can't see that in logs with a ton of pain

But I'm also fine with merging this in and splitting it up later

**Comment by mfield** [comment]
attempt 1 at splitting: i do kind of like how simple the separate boxes are. i don't like how physically separated they are. is there a way to force them next to or on top of each other?
image

**Comment by abelito75** [comment]
I don't think so 
But if that is okay we cna merge it in

**Comment by mfield** [comment]
yep let's send it then

**Pull request merged**